src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -46,43 +56,153 @@ const useCoreChatSystemHook = () => {
+      return findMember?.user?.image;
+    }
+    return DEFAULT_PROFILE_PIC_PATH;
+  };
 
-  const saveChannelListData = async () => {
-    if (!localDb) return;
+  const onSignedPostNotifReceived: (data: GetstreamFeedListenerObject) => Promise<void> = async (
+    data
+  ) => {
+    try {
+      const activityId = data?.new[0]?.object?.id || data?.new[0]?.id;
+      await getSingleSignedPostNotification(activityId);
+    } catch (e) {
+      console.log('error getSingleSignedPostNotification');
+      console.log(e);
+    }
+  };
+
+  usePostNotificationListenerHook(onAnonPostNotifReceived, true);
+  usePostNotificationListenerHook(onSignedPostNotifReceived, false);
+
+  const {lastAnonymousMessage, lastSignedMessage} = useBetterWebsocketHook();
+
+  const getUnreadCount = (message: GetstreamMessage, selectedChannel): number => {
+    const isMyMessage =
+      message?.user?.id === signedProfileId || message?.user?.id === anonProfileId;
+
+    const hasUnreadMessage = selectedChannel?.unread_count <= 0;
+    if (isMyMessage) return 0;
+    if (hasUnreadMessage) return selectedChannel?.unread_count + 1;
+    return 1;
+  };
+
+  const helperSignedChannelListData = async (websocketData: GetstreamWebsocket) => {
+    const {type} = websocketData;
+    const websocketMessage = websocketData?.message;
+
+    const chatName = await getChatName(websocketData?.channel?.name, profile?.username);
+    if (websocketData?.channel?.anon_user_info_emoji_name) {
+      websocketData.targetName = `Anonymous ${websocketData?.channel?.anon_user_info_emoji_name}`;
+    } else {
+      websocketData.targetName = chatName;
+    }
+
+    let selectedChannel;
+    try {
+      selectedChannel = (await ChannelList.getById(localDb, websocketData?.channel_id)) as any;
+    } catch (error) {
+      console.log('error on getting selectedChannel');
+    }
+
+    websocketData.targetImage = handleChannelImage(websocketData?.channel?.members);
+    const websocketChannelType = websocketData?.channel_type;
+    if (websocketChannelType === 'topics' || websocketChannelType === 'group') {
+      websocketData.targetImage = websocketData?.channel?.channel_image;
+    } else {
+      websocketData.targetImage = selectedChannel?.channel_picture ?? websocketMessage?.user?.image;
+    }
+
+    if (type === 'notification.added_to_channel') {
+      websocketData.message.text = 'You have been added to this group';
+    }
+    if (type === 'notification.removed_from_channel') {
+      websocketData.message.text = 'You have been removed from this group';
+    }
+
+    websocketData.unread_count = getUnreadCount(websocketMessage, selectedChannel);
+    websocketData.anon_user_info_color_name = websocketData?.channel?.anon_user_info_color_name;
+    websocketData.anon_user_info_emoji_code = websocketData?.channel?.anon_user_info_emoji_code;
+    websocketData.anon_user_info_color_code = websocketData?.channel?.anon_user_info_color_code;
+
+    return websocketData;
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `getUnreadCount` function, the condition `selectedChannel?.unread_count <= 0` seems to be incorrect. It should be `selectedChannel?.unread_count > 0` because we want to increment the unread count only when there are already some unread messages.
```diff
-    const hasUnreadMessage = selectedChannel?.unread_count <= 0;
+    const hasUnreadMessage = selectedChannel?.unread_count > 0;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->